community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/ast/convert/commands/ExpressionConverters.scala

```diff
@@ -47,6 +47,8 @@ object ExpressionConverters {
       case e: ast.NotEquals => e.asCommandNotEquals
       case e: ast.RegexMatch => e.asCommandRegex
       case e: ast.In => e.asCommandIn
+      case e: ast.Like => e.asCommandRegex
+      case e: ast.NotLike => e.asCommandNegatedRegex
```

Comment:
I am not against having `LIKE` and `NOT LIKE` at the AST level, but before planning we should remove the syntactic sugar as we do for `Equals`. I am not sure I got the same-generate-expression part of your comment since I have in mind the classes `ast.Like` and `ast.NotLike` as by diff above.